### PR TITLE
Fix: do not double-wrap OpenSearchException.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Before:

```
org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [http_exception] server returned 400
    at org.opensearch.client.transport.aws.AwsSdk2Transport.parseResponse (AwsSdk2Transport.java:542)
    at org.opensearch.client.transport.aws.AwsSdk2Transport.executeSync (AwsSdk2Transport.java:442)
    at org.opensearch.client.transport.aws.AwsSdk2Transport.performRequest (AwsSdk2Transport.java:246)
    at org.opensearch.client.opensearch.indices.OpenSearchIndicesClient.create (OpenSearchIndicesClient.java:248)
```

After:

```
org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [resource_already_exists_exception] OpenSearch exception [type=resource_already_exists_exception, reason=index [movies/u0Qdo4UBHwS29ruw8Xao] already exists]
    at org.opensearch.client.transport.aws.AwsSdk2Transport.parseResponse (AwsSdk2Transport.java:534)
    at org.opensearch.client.transport.aws.AwsSdk2Transport.executeSync (AwsSdk2Transport.java:442)
    at org.opensearch.client.transport.aws.AwsSdk2Transport.performRequest (AwsSdk2Transport.java:246)
    at org.opensearch.client.opensearch.indices.OpenSearchIndicesClient.create (OpenSearchIndicesClient.java:248)
```

~Help me write a unit test for this? Is it worth it?~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java [hunk 1]
```diff
@@ -456,6 +457,8 @@ private <ResponseT, ErrorT> ResponseT parseResponse(
                     ErrorT error = errorDeserializer.deserialize(parser, mapper);
                     throw new OpenSearchException((ErrorResponse) error);
                 }
+            } catch (OpenSearchException e) {
```
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java:460]: Would be good to cover that with tests ... :-)

**Review by reta — APPROVED** [review]

File: java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java (hunk 1 above)
Comment (dblock) [reply java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java:459]: Added integration test.

**Review by reta — APPROVED** [review]

**Comment by dblock** [comment]
@VachaShah care for a 2PR? thanks!

**Review by VachaShah — DISMISSED** [review]

**Review by reta — APPROVED** [review]

**Comment by dblock** [comment]
@VachaShah need one more pls

**Review by VachaShah — APPROVED** [review]

**Pull request merged**